app/models/scheme.rb

```diff
@@ -295,6 +295,16 @@ def active?
     status == :active
   end
 
+  def all_locations_inactive?
+    active? && locations.all? { |location| !location.active? }
+  end
```

Comment:
I'd probably flip this and call it something like `has_active_locations?` cause it reads more clearly I'd also use scopes here cause that's faster in theory 
so something like `locations.active.exists?`

Reply:
Okay good idea